applications/frontier/src/frontier_handle_acl.erl

```diff
@@ -0,0 +1,129 @@
+    Entity = wh_json:get_value(<<"Entity">>, Reqest),
+    IncludeRealm  = wh_json:is_true(<<"With-Realm">>, Reqest, 'false'),
+    Payload = lookup_acl_records(Entity, IncludeRealm),
+    send_response(Reqest, Payload).
+
+-spec send_response(wh_json:object(), wh_json:objects()) -> any().
+send_response(Reqest, Responses) ->
+    RespStub = wh_json:from_list([{<<"Msg-ID">>, wh_json:get_value(<<"Msg-ID">>, Reqest)}
+                                  | wh_api:default_headers(?APP_NAME, ?APP_VERSION)
+                                 ]),
+    ServerID = wh_json:get_value(<<"Server-ID">>, Reqest),
+    {DeviceACLs, RealmACLs} = lists:partition(fun frontier_utils:is_device/1, Responses),
+    Resp = lists:foldl(fun wh_json:merge_jobjs/2, RespStub, [make_section(DeviceACLs, <<"Device">>)
+                                                             ,make_section(RealmACLs, <<"Realm">>)
+                                                            ]),
+    lager:debug("publishing response"),
+    wapi_frontier:publish_acls_resp(ServerID, Resp).
+
+-spec make_section(wh_json:objects(), ne_binary()) -> wh_json:object().
+-spec make_section(ne_binary(), api_binary(), api_binaries(), api_binary()) -> wh_json:object().
+make_section([], _) ->
+    wh_json:new();
+make_section([JObj], Section) ->
+    Order = wh_json:get_value([<<"value">>, <<"acls">>, <<"order">>], JObj),
+    CIDRs = wh_json:get_value([<<"value">>, <<"acls">>, <<"cidrs">>], JObj),
+    UserAgent = wh_json:get_value([<<"value">>, <<"acls">>, <<"user_agent">>], JObj),
+    make_section(Section, Order, CIDRs, UserAgent).
+make_section(_, Order, CIDRs, _) when Order =:= 'undefined'
+                                      orelse CIDRs =:= 'undefined' ->
+        wh_json:new();
+make_section(Section, Order, CIDRs, UserAgent) ->
+    Props = props:filter_undefined([{<<"Order">>, Order}
+                                    ,{<<"CIDR">>, CIDRs}
+                                    ,{<<"User-Agent">>, UserAgent}
+                                   ]),
+    wh_json:from_list([{Section, wh_json:from_list(Props)}]).
+
+-spec lookup_acl_records(ne_binary(), boolean()) -> wh_json:objects().
+lookup_acl_records(Entity, IncludeRealm) ->
+    lager:debug("Handle acl request for ~s", [Entity]),
+    Realm = frontier_utils:extract_realm(Entity),
+    case whapps_util:get_account_by_realm(Realm) of
+        {'ok', _} ->
+            lager:debug("Found realm, try to send response"),
+            run_acl_query(Entity, IncludeRealm);
+        _ ->
+            lager:info("Can't find realm ~s. Sending deny ACL.", [Realm]),
+            make_deny_acl(Entity, IncludeRealm)
+    end.
+
+-spec lookup_acl_records(ne_binary()) -> wh_json:objects().
+lookup_acl_records(Entity) ->
+    lookup_acl_records(Entity, 'true').
+
+-spec run_acl_query(ne_binary(), boolean()) -> wh_json:objects().
+run_acl_query(Entity, IncludeRealm) ->
+    Keys = binary:split(Entity, <<"@">>),
+    ViewOpts = case Keys of
```

Comment:
Put this in a function `build_view_options(Entity, IncludeRealm)` and do the splitting of Entity in there.